js.go

```diff
@@ -1413,6 +1430,14 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync,
 		return nil, err
 	}
 
+	// Wait for context to get canceled if there is one.
```

Comment:
You should move this block at the end of the function, on success. You will note that at around line 1530 (or 1505 of original code), the subscription may be recreated.